x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java

```diff
@@ -241,6 +244,30 @@ public boolean isActionSafe(StepKey stepKey) {
         }
     }
 
+    /**
+     * Validate the name for an policy against some static rules. Intended to match
+     * {@link org.elasticsearch.cluster.metadata.MetaDataCreateIndexService#validateIndexOrAliasName(String, BiFunction)}
+     * @param policy the policy name to validate
+     * @throws IllegalArgumentException if the name is invalid
+     */
+    public static void validatePolicyName(String policy) {
+        if (policy.contains(",")) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not contain ','");
+        }
+        if (policy.contains(" ")) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not contain spaces");
+        }
+        if (policy.charAt(0) == '_') {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not start with '_', '-', or '+'");
+        }
```

Comment:
I wonder if it would be nicer for the user if we just have a single message here that describes all the characters that we don't allow in a policy name? This might avoid frustration if the user tries something like `_foo bar` as the policy name since they will be able to fix everything in one step rather than two?

Reply:
I was thinking it would be ok as 
```
"invalid policy name [" + policy + "]: must not start with '_', '-', or '+'" and must not contain any spaces or `,`."
```
But I don't feel too strongly about this, I agree the likelihood of multiple violations is low since there are too many rules here

Reply:
I would argue that's harder to consume for the case of a single violation (the common case contingent on any violation) as now as a user I am left wondering, wait, which rule did I break?

It's like those awful password dialog boxes. Password doesn't meet the required criteria:
 - must be between 8 and 16 characters on Monday--Friday
 - must be between 7 and 17 characters on weekends
 - contains none of !@*&#
 - contains at least two of ()_+-
 - contains at least four digits
 - contains at least one upper-case letter
 - contains at least two lower-case letters
 - not contain three of the same character
 - not contain any adjacent repeating characters
 - not contain adjacent consecutive digits or letters
 - unless in reverse order

Reply:
ok lets leave it as is then

Reply:
1. I'm on board with leaving this as separate messages.
2. This particular message was out of date, I've updated it to only say that policy names cannot start with `_` (the other characters are okay).